src/templates/addons/aws/modules/core/region.ts

```diff
@@ -0,0 +1,24 @@
+import { dedent } from 'ts-dedent';
+
+import { appendToFile } from '@/helpers/file';
+import { AwsOptions } from '@/templates/addons/aws';
+import { AWS_DEFAULT_REGION } from '@/templates/addons/aws/constants';
+import { INFRA_BASE_VARIABLES_PATH } from '@/templates/core/constants';
+
+const regionVariablesContent = (awsRegion: string) => dedent`
+  variable "region" {
+    description = "AWS region"
+    type        = string
+    default     = "${awsRegion}"
+  }`;
+
+const applyAwsRegion = async (options: AwsOptions) => {
+  appendToFile(
+    INFRA_BASE_VARIABLES_PATH,
+    regionVariablesContent(options.awsRegion || AWS_DEFAULT_REGION),
+    options.projectName
+  );
+};
```

Comment:
we need to copy the region variable to the the "SHARED" workspace as well, otherwise, it will fail when run the plan as the var is missing
<img width="738" alt="Screenshot 2023-08-17 at 11 23 29" src="https://github.com/nimblehq/infrastructure-templates/assets/1896814/9b76db6e-6590-46aa-ae30-4f861fb2fdfe">

Reply:
Good catch, added in https://github.com/nimblehq/infrastructure-templates/pull/169/commits/1d3f8787825086d882e596a77cbf57248b278449